pkg/kubelet/types/pod_update.go

```diff
@@ -147,6 +147,21 @@ func IsCriticalPod(pod *v1.Pod) bool {
 	return IsCritical(pod.Namespace, pod.Annotations) || (pod.Spec.Priority != nil && IsCriticalPodBasedOnPriority(*pod.Spec.Priority))
 }
 
+// Preemptable returns true if preemptor pod can preempt preemptee pod:
+//   - If preemptee's priority is greater than or equal to SystemCriticalPriority,
+//     preemptee can not be preempted (return false)
+//   - If preemptor's is greater than preemptee's priority, it's preemptable (return true)
+//   - If preemptor is nill and preemptee bears the critical pod annotation key,
+//     it can not be preempted (return false)
+func Preemptable(preemptor, preemptee *v1.Pod) bool {
+	if (preemptor != nil && preemptor.Spec.Priority != nil) &&
+		(preemptee != nil && preemptee.Spec.Priority != nil) {
+		return *(preemptor.Spec.Priority) > *(preemptee.Spec.Priority)
```

Comment:
I think this will still allow a `node-system-critical` pod to preempt a `cluster-system-critical` pod, which doesn't match what the function description says.

Reply:
> This will still allow a node-system-critical pod to preempt a cluster-system-critical pod

That's what we want: let `node-system-critical` preempt `cluster-system-critical`. Check Priority instead of annotation if both pod has priority.

Reply:
can you update the function description?  The first bullet is that system-critical pods cannot be preempted.

Reply:
The logic is correct to me, but I agree with dashpole that comment the should be changed.

Reply:
oh, yes, done